esrally/client.py

```diff
@@ -12,7 +12,12 @@ class EsClientFactory:
     Abstracts how the Elasticsearch client is created. Intended for testing.
     """
     def __init__(self, hosts, client_options):
-        logger.info("Creating ES client connected to %s with options [%s]" % (hosts, client_options))
+        masked_client_options = dict(client_options)
+        if masked_client_options.has_key("basic_auth_password"):
+            masked_client_options["basic_auth_password"] = "*****"
+        if masked_client_options.has_key("http_auth"):
+            masked_client_options["http_auth"] = (client_options["basic_auth_user"], "*****")
```

Comment:
I think this should be `masked_client_options["http_auth"] = (client_options["http_auth"][0], "*****")` as you are modifying the `http_auth` entry and a `basic_auth_user` does not necessarily exist then.